src/XMLimport.cpp

```diff
@@ -1952,162 +1505,139 @@ void XMLimport::readKeyPackage()
             }
-            else if (name() == "filepath"){
+            else if( name() == "filepath" ) {
                 entry << readElementText();
-                //map[key] = readElementText();
             }
-            else if (name() == "globalSave"){
+            else if( name() == "globalSave" ) {
                 entry << readElementText();
             }
-            else if (name() == "priority"){
+            else if( name() == "priority" ) {
+                // The last expected detail for the entry - so store this completed entry into the QMap
                 entry << readElementText();
                 map[key] = entry;
                 entry.clear();
-
             }
-            else
-            {
+            else {
                 readUnknownHostElement();
             }
         }
     }
 }
 
-
 void XMLimport::readStringList( QStringList & list )
 {
-    while( ! atEnd() )
-    {
-        readNext();
+    while( ! atEnd() ) {
 
-        if( isEndElement() ) break;
+        readNext();
 
-        if( isStartElement() )
-        {
-            if( name() == "string")
-            {
+        if( isEndElement() ) {
+            break;
+        }
+        else if( isStartElement() ) {
+            if( name() == "string" ) {
                 list << readElementText();
             }
-            else
-            {
+            else {
                 readUnknownTriggerElement();
             }
         }
     }
 }
 
-void XMLimport::readIntegerList( QList<int> & list )
+void XMLimport::readIntegerList( QList<int> & list, const QString & parentName )
 {
-    while( ! atEnd() )
-    {
-        readNext();
+    while( ! atEnd() ) {
 
-        if( isEndElement() ) break;
+        readNext();
 
-        if( isStartElement() )
-        {
-            if( name() == "integer")
-            {
+        if( isEndElement() ) {
+            break;
+        }
+        else if( isStartElement() ) {
+            if( name() == "integer" ) {
+                QString numberText = readElementText();
                 bool ok = false;
-                int num;
-                num = readElementText().toInt( &ok, 10 );
-                if( ok )
-                {
+                int num = numberText.toInt( &ok, 10 );
+                if( Q_LIKELY( ! numberText.isEmpty() && ok ) ) {
                     list << num;
                 }
-                else
-                {
-                    qFatal("FATAL ERROR: reading package property list contained invalid elements");
+                else {
+                    // Using qFatal() seems a little, erm, fatalistic but
+                    // it seems no lesser one will always be detectable on the
+                    // RELEASE version on Windows? - Slysven
+                    qFatal( "XMLimport::readIntegerList(...) ERROR: unable to convert: \"%1\" to a number when reading the 'regexCodePropertyList' element of the 'Trigger' or 'TriggerGroup' element!",
```

Comment:
:beetle: Bother should have used `%s` not `%1` as it is a `printf()` format specifier not a `QString::arg()` replacement value - also missing a second `%s` for the `parentName`.